miniapps/fluids/schrodinger_flow.hpp

```diff
@@ -0,0 +1,947 @@
+   MFEM_HOST_DEVICE Complex(real_t r) { x = r, y = 0.0; }
+   MFEM_HOST_DEVICE Complex(real_t r, real_t i) { x = r, y = i; }
+   MFEM_HOST_DEVICE real_t real() const { return x; }
+   MFEM_HOST_DEVICE void real(real_t r) { x = r; }
+   MFEM_HOST_DEVICE real_t imag() const { return y; }
+   MFEM_HOST_DEVICE void imag(real_t i) { y = i; }
+
+   template <typename U>
+   MFEM_HOST_DEVICE inline Complex &operator*=(const U &z)
+   {
+      return (*this = *this * z, *this);
+   }
+
+   template <typename U>
+   MFEM_HOST_DEVICE inline Complex &operator/=(const U &z)
+   {
+      return (*this = *this / z, *this);
+   }
+};
+
+MFEM_HOST_DEVICE inline Complex operator*(const Complex &x, const real_t &y)
+{
+   return Complex(x.real() * y, x.imag() * y);
+}
+
+MFEM_HOST_DEVICE inline Complex operator+(const Complex &a, const Complex &b)
+{
+   return Complex(a.real() + b.real(), a.imag() + b.imag());
+}
+
+MFEM_HOST_DEVICE inline Complex operator*(const real_t d, const Complex &z)
+{
+   return Complex(z.real() * d, z.imag() * d);
+}
+
+MFEM_HOST_DEVICE inline Complex operator*(const Complex &a, const Complex &b)
+{
+   return Complex(a.real() * b.real() - a.imag() * b.imag(),
+                  a.real() * b.imag() + a.imag() * b.real());
+}
+
+MFEM_HOST_DEVICE inline Complex operator/(const Complex &z, const real_t &d)
+{
+   return Complex(z.real() / d, z.imag() / d);
+}
+
+MFEM_HOST_DEVICE inline real_t abs(const Complex &z)
+{
+   return std::hypot(z.real(), z.imag());
+}
+
+MFEM_HOST_DEVICE inline Complex exp(const Complex &q)
+{
+   real_t s, c, e = std::exp(q.real());
+#ifdef MFEM_USE_SINGLE
+   sincosf(q.imag(), &s, &c);
+#else
+   sincos(q.imag(), &s, &c);
+#endif
+   return Complex(c * e, s * e);
+}
+
+MFEM_HOST_DEVICE inline real_t norm(const Complex &z)
+{
+   return z.real() * z.real() + z.imag() * z.imag();
+}
+
+using complex_t = Complex;
+#endif // MFEM_USE_CUDA || MFEM_USE_HIP
+
+namespace mfem
+{
+
+using real3_t = std::array<real_t, 3>;
+
+/// @brief Base class for Schrodinger solver kernels.
+template <typename TMesh,
+          typename TFiniteElementSpace,
+          typename TComplexGridFunction,
+          typename TGridFunction,
+          typename TBilinearForm,
+          typename TMixedBilinearForm,
+          typename TLinearForm>
+struct SchrodingerBaseKernels: public Options
+{
+   std::function<Mesh()> CreateMesh2D, CreateMesh3D;
+   Mesh serial_mesh;
+   TMesh mesh;
+   H1_FECollection h1_fec;
+   ND_FECollection nd_fec;
+   TFiniteElementSpace h1_fes, nd_fes, nodal_fes;
+   TGridFunction nodes;
+   const int ne, ndofs;
+   ConstantCoefficient one;
+   VectorFunctionCoefficient Vx, Vy, Vz;
+   TBilinearForm mass_h1, mass_nd, diff_h1;
+   TMixedBilinearForm grad_nd, nd_dot_x_h1, nd_dot_y_h1, nd_dot_z_h1;
+   Array<int> ess_tdof_list;
+   std::function<void()> SetEssentialTrueDofs;
+   bool ess_tdof_list_setup, diff_h1_setup;
+   OperatorJacobiSmoother Km1_h1_smoother;
+   OrthoSolver Km1_ortho;
```

Comment:
For code readability 
```suggestion
   OrthoSolver diff_h1_ortho;
```

Reply:
done in https://github.com/mfem/mfem/pull/5059/commits/e6c9f73fc6483af5c98055b65a0652b164e0e354